packages/mattermost-redux/src/reducers/entities/work_templates.ts

```diff
@@ -57,9 +57,23 @@ function playbookTemplates(state: [] = [], action: GenericAction) {
     }
 }
 
+function linkedProducts(state: Record<string, number> = {}, action: GenericAction) {
+    switch (action.type) {
+    case WorkTemplatesType.EXECUTE_SUCCESS: {
+        return {
```

Comment:
Since the state we're modelling is the counts of what was linked in the work template that was just executed, I think it would be a little more robust to `return action.data` instead of merging it with the existing state. If the implementation of `getContentCount` were altered in the future so that only items with a positive count were in the object, or other keys were only conditionally there, then you could get stale data where a first work template run that had boards and the second one did not have boards, you would keep seeing the boards UI even though it was relevant to the second work template execution.

Reply:
the way it is implemented right now is that in every successful execution that data model is overwritten, so on every other worktemplate use, the data in the global store contains the information for the new template included items.